kitt_bar_app/ticket.rb

```diff
@@ -0,0 +1,89 @@
+class Ticket
+  attr_reader :content,
+              :table,
+              :student,
+              :teacher,
+              :exercice_name,
+              :current_user_can_take,
+              :current_user_can_leave,
+              :current_user_can_update,
+              :current_user_can_cancel,
+              :current_user_can_mark_as_solved,
+              :api_v1_ticket_path,
+              :cancel_api_v1_ticket_path,
+              :mark_as_solved_api_v1_ticket_path,
+              :leave_api_v1_ticket_path,
+              :take_api_v1_ticket_path
+
+  def initialize(attr = {})
+    @content = attr["content"]
+    @table = attr["table"]
+    @student = attr["user"]["name"]
+    @teacher =  attr["assigned"]["name"]  if attr["assigned"]
+    @is_mine = attr["is_mine"]
+    @remote = attr["remote"]
+    @exercice_name = attr["exercice_name"]
+    @owner_slack_uid = attr["owner_slack_uid"]
+    @slack_team_id = attr["slack_team_id"]
+    @current_user_can_take = attr["policy"]["current_user_can_take"]
+    @current_user_can_leave = attr["policy"]["current_user_can_leave"]
+    @current_user_can_update = attr["policy"]["current_user_can_update"]
+    @current_user_can_cancel = attr["policy"]["current_user_can_cancel"]
+    @current_user_can_mark_as_solved = attr["policy"]["current_user_can_mark_as_solved"]
+    @api_v1_ticket_path = attr["routes"]["api_v1_ticket_path"]
+    @cancel_api_v1_ticket_path = attr["routes"]["cancel_api_v1_ticket_path"]
+    @mark_as_solved_api_v1_ticket_path = attr["routes"]["mark_as_solved_api_v1_ticket_path"]
+    @leave_api_v1_ticket_path = attr["routes"]["leave_api_v1_ticket_path"]
+    @take_api_v1_ticket_path = attr["routes"]["take_api_v1_ticket_path"]
```

Comment:
all these could be digged instead? I don't know if we should favoritise `dig` instead of `.[]` wdyt? 🤔

Reply:
Feels like `.dig` is safer and "more forgiving" .. even though we'll probably always have "policy" when we have a ticket. Going for a `.dig` everywhere is also easier to read imo. Will push a new version with digs instead of []